### PR TITLE
[Java] Added JNI for getMapValueForKeys

### LINKED ISSUE #10818: [FEA] JNI bindings for `maps_column_view::get_value_for(column_view)`
#10380 introduced a `maps_column_view` abstraction to interpret `LIST<STRUCT<key,value>>` columns as maps, and look up values corresponding to keys specified either as `scalar`s or `column`s.

However, the JNI layer currently supports `ColumnView::getMapValue()` using only Scalar keys. To support key lookup using per-row expression values, one would need a  `ColumnView::getMapValueFromKeyVector()` that takes a `ColumnView`. This is required to support queries like those specified in https://github.com/NVIDIA/spark-rapids/issues/5204.

Note that the heavy lifting is already done in the native code, i.e. `maps_column_view::get_value_for(column_view)`. All that is required is the JNI plumbing to patch through to `maps_column_view`.

### PR DESCRIPTION
**Pull request by razajafri**
This PR adds Java method for getting values for a list of keys

fixes #10818

### CODE REVIEW

**Comment by res-life** [comment]
Seems need to format C++ code.

```
cd cudf
pre-commit run --all-files
...
clang-format.............................................................Failed
...
```
refer to https://github.com/rapidsai/cudf/blob/branch-22.04/CONTRIBUTING.md#pre-commit-hooks

File: java/src/main/java/ai/rapids/cudf/ColumnView.java [hunk 1]
```diff
@@ -3250,6 +3250,19 @@ private static void assertIsSupportedMapKeyType(DType keyType) {
...
   }
 
+  /**
+   * Given a column of type List<Struct<X, Y>> and a key of type X, return a column of type Y,
+   * where each row in the output column is the Y value corresponding to the X key.
+   * If the key is not found, the corresponding output value is null.
+   * @param keys the column view with keys to lookup in the column
+   * @return a column of values or nulls based on the lookup result
+   */
+  public final ColumnVector getMapValueForKeys(ColumnView keys) {
```
Comment (revans2) [inline java/src/main/java/ai/rapids/cudf/ColumnView.java:3260]: Why not just call it `getMapValue` just like the Scalar API is called? If they do the same thing, then to me it is ideal to call them the same thing.

**Review by revans2 — APPROVED** [review]
Just a nit.

**Comment by razajafri** [comment]
rerun tests

**Review by revans2 — APPROVED** [review]

**Comment by razajafri** [comment]
rerun tests

**Comment by razajafri** [comment]
rerun tests

**Comment by razajafri** [comment]
@gpucibot merge

**Pull request merged**